internal/pkg/config/config.go

```diff
@@ -284,27 +284,59 @@ func (fm funcMatcher) MatchFunction(path, receiver, name string) bool {
 	return fm.Package.MatchString(path) && fm.Receiver.MatchString(receiver) && fm.Method.MatchString(name)
 }
 
-var readFileOnce sync.Once
-var readConfigCached *Config
-var readConfigCachedErr error
+// readConfigResults reduces disk access across multiple ReadConfig calls.
+type readConfigResults struct {
+	once sync.Once
+	conf *Config
+	err  error
+}
 
-// ReadConfig reads, parses, and validates config file.
-func ReadConfig() (*Config, error) {
-	readFileOnce.Do(func() {
+// configCache safely stores readConfigResults for concurrent access.
+type configCache struct {
+	mu    sync.Mutex
+	cache map[string]*readConfigResults
+}
+
+// Instantiates readConfigResults if file has not yet been loaded
+func (w *configCache) read(file string) (*Config, error) {
+	w.instantiateCacheForFile(file)
+
+	cc := w.cache[file]
+	cc.once.Do(func() {
 		c := new(Config)
 		bytes, err := ioutil.ReadFile(configFile)
 		if err != nil {
-			readConfigCachedErr = fmt.Errorf("error reading analysis config: %v", err)
+			cc.err = fmt.Errorf("error reading analysis config: %v", err)
 			return
 		}
 
 		if err := yaml.UnmarshalStrict(bytes, c); err != nil {
-			readConfigCachedErr = err
+			cc.err = err
 			return
 		}
-		readConfigCached = c
+		cc.conf = c
 	})
-	return readConfigCached, readConfigCachedErr
+
+	return cc.conf, cc.err
+}
+
+func (w *configCache) instantiateCacheForFile(file string) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	if w.cache[file] == nil {
+		w.cache[file] = new(readConfigResults)
+	}
+}
+
+var cache = configCache{
+	mu:    sync.Mutex{},
+	cache: make(map[string]*readConfigResults),
+}
+
+// ReadConfig reads, parses, and validates config file.
+func ReadConfig() (*Config, error) {
```

Comment:
Nit: consider moving this function above the others so that the code can be read top-to-bottom.

Reply:
Done.